torch/distributed/_shard/sharded_tensor/_ops/matrix_ops.py

```
@@ -0,0 +1,330 @@
+    sharding_dim = st.sharding_spec().dim  # type: ignore[attr-defined]
+    narrow_idx = None
+    for idx in range(-1, -mask.dim() - 1, -1):
+        if st.dim() + idx == sharding_dim and mask.size(idx) != 1:
+            narrow_idx = idx
+    if narrow_idx is not None:
+        rank_idx = None
+        for idx, placement in enumerate(st._sharding_spec.placements):  # type: ignore[attr-defined]
+            if placement.rank() == current_rank:  # type: ignore[index]
+                rank_idx = idx  # type: ignore[attr-defined]
+        shard_metadata = st.metadata().shards_metadata[rank_idx]  # type: ignore[index]
+        mask = mask.narrow(
+            narrow_idx,
+            shard_metadata.shard_offsets[sharding_dim],
+            shard_metadata.shard_sizes[sharding_dim],
+        )
+    local_tensor = st.local_tensor().masked_fill(mask, value)
+    return local_tensor, st.sharding_spec(), st.size()
+
+
+_register_sharded_op_on_local_tensor(
+    torch.Tensor.masked_fill,
+    extra_check=sharded_masked_fill_check,
+    customized_func=sharded_masked_fill,
+)
+
+
+def sharded_view_check(*args, **kwargs):
+    """
+    Perform extra checks for the sharded_view op.
+
+    Args: same as ``torch.Tensor.view``.
+
+    Return: None
+    """
+    st = args[0]
+    shapes = args[1:]
+    if len(shapes) == 0:
+        raise ValueError("Missing *shape for sharded view op.")
+    if len(shapes) <= st.sharding_spec().dim:
+        raise NotImplementedError(
+            f"Shape having dim {len(shapes)} is not supported "
+            f"for sharded tensor sharded on dim {st.sharding_spec().dim}."
+        )
+    st_size = math.prod(st.size())  # type: ignore[attr-defined]
+    shape_size = math.prod(shapes)  # type: ignore[attr-defined]
+    neg_sum = sum(i for i in shapes if i < 0)
+    if shape_size > st_size or st_size % shape_size:
+        raise ValueError("Shape is invalid for sharded tensor size.")
```

maybe let's throw the same error as the torch.Tensor case?